src/typescript/typescript.ts

```diff
@@ -446,7 +609,42 @@ export class TypeScriptProject extends NodeProject {
       `<rootDir>/(${this.testdir}|${this.srcdir})/**/*(*.)@(spec|test).ts?(x)`
     );
 
+    const jestMajorVersion = semver.coerce(jest.jestVersion)?.major;
     // add relevant deps
+    if (!jestMajorVersion || jestMajorVersion >= 29) {
+      return this.addJestNoCompileModern(jest, tsJestOptions);
+    }
+    this.addJestNoCompileLegacy(jest, tsJestOptions);
+  }
+
+  private addJestNoCompileModern(
+    jest: Jest,
+    tsJestOptions: TsJestOptions | undefined
+  ) {
+    jest.config.transform = deepMerge([
+      {
+        [tsJestOptions?.tranformPattern ??
+        TypeScriptProject.DEFAULT_TS_JEST_TRANFORM_PATTERN]: new Transform(
+          "ts-jest",
+          {
+            tsconfig: this.tsconfigDev.fileName,
```

Comment:
Should this technically now be this?
```suggestion
            tsconfig: TsJestTsconfig.fromFile(this.tsconfigDev.fileName),
```

Reply:
Yeah, definitely, wondering how TS didn't catch, but I guess it has to be something about the return types.